hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/SimpleKeyGenerator.java

```diff
@@ -36,8 +36,11 @@
   private final SimpleAvroKeyGenerator simpleAvroKeyGenerator;
 
   public SimpleKeyGenerator(TypedProperties props) {
-    this(props, props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key()),
-        props.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key()));
+    this(props,
+        (props.containsKey(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key())
```

Comment:
getString(String property, String defaultValue) is also available in  TypedProperties

Reply:
fixed.